src/main/java/com/ensiasit/projectx/services/TeamServiceImpl.java

```diff
@@ -0,0 +1,139 @@
+package com.ensiasit.projectx.services;
+
+import com.ensiasit.projectx.dto.TeamDto;
+import com.ensiasit.projectx.exceptions.BadRequestException;
+import com.ensiasit.projectx.exceptions.NotFoundException;
+import com.ensiasit.projectx.models.Affiliation;
+import com.ensiasit.projectx.models.Contest;
+import com.ensiasit.projectx.models.Team;
+import com.ensiasit.projectx.repositories.AffiliationRepository;
+import com.ensiasit.projectx.repositories.ContestRepository;
+import com.ensiasit.projectx.repositories.TeamRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class TeamServiceImpl implements TeamService {
+    private final TeamRepository teamRepository;
+    private final ContestRepository contestRepository;
+    private final AffiliationRepository affiliationRepository;
+
+    @Override
+    public List<TeamDto> getAll() {
+        return teamRepository.findAll().stream()
+                .map(team -> TeamDto.builder()
+                        .id(team.getId())
+                        .name(team.getName())
+                        .contestId(team.getContest().getId())
+                        .affiliationId(team.getAffiliation().getId())
+                        .affiliationName(team.getAffiliation().getName())
+                        .build())
+                .toList();
+    }
+
+    @Override
+    public TeamDto createTeam(TeamDto teamDto) {
+        if (teamRepository.existsByName(teamDto.getName())) {
+            throw new BadRequestException("Name already taken");
+        }
+
+        Optional<Contest> optionalContest = contestRepository.findById(teamDto.getContestId());
+
+        if (optionalContest.isEmpty()) {
+            throw new BadRequestException("Incorrect contest id");
+        }
+
+        Optional<Affiliation> optionalAffiliation = affiliationRepository.findById(teamDto.getAffiliationId());
+
+        if (optionalAffiliation.isEmpty()) {
```

Comment:
The same here. I think we should decide if we handle database constraints here or let `Hibernate` do the job